public/lib/monaco/json.ts

```diff
@@ -0,0 +1,56 @@
+import { monaco } from "@osd/monaco";
+// @ts-ignore
+import * as mode from "monaco-editor/esm/vs/language/json/jsonMode.js"; // eslint-disable-line
+
+const Emitter = monaco.Emitter;
+class LanguageServiceDefaultsImpl {
+  _onDidChange: monaco.Emitter<any>;
+  _languageId: string;
+  _diagnosticsOptions!: monaco.languages.json.DiagnosticsOptions;
+  constructor(languageId: string, diagnosticsOptions: monaco.languages.json.DiagnosticsOptions) {
+    this._onDidChange = new Emitter();
+    this._languageId = languageId;
+    this.setDiagnosticsOptions(diagnosticsOptions);
+  }
+  public get languageId() {
+    return this._languageId;
+  }
+  public get onDidChange() {
+    return this._onDidChange.event;
+  }
+  public get diagnosticsOptions() {
+    return this._diagnosticsOptions;
+  }
+  setDiagnosticsOptions(options: monaco.languages.json.DiagnosticsOptions) {
+    this._diagnosticsOptions = options || Object.create(null);
+    this._onDidChange.fire(this);
+  }
+}
+export { LanguageServiceDefaultsImpl };
+const diagnosticDefault = {
+  validate: true,
+  allowComments: true,
+  schemas: [],
+  enableSchemaRequest: false,
+};
+/**
+ * In case there are other plugins register the language implementation.
+ */
+if (!monaco.languages.json) {
+  const jsonDefaults = new LanguageServiceDefaultsImpl("json", diagnosticDefault);
+  function createAPI() {
+    return {
+      jsonDefaults,
+    };
+  }
+  monaco.languages.json = createAPI();
+  monaco.languages.register({
+    id: "json",
+    extensions: [".json", ".bowerrc", ".jshintrc", ".jscsrc", ".eslintrc", ".babelrc"],
```

Comment:
have we used the extensions?

Reply:
There are two ways the editor will judge which language to use, one is declare the language explicitly <MonacoEditor 
 language="json" />, the other way is that it will assume the language according to the model extension(you can consider a model as a file). We are not using the extensions but it is a official implement in Monaco editor repo. We'd better keep it.